test/unit/org/apache/cassandra/index/sai/disk/v1/InvertedIndexSearcherTest.java

```diff
@@ -220,6 +220,8 @@ private List<Pair<ByteComparable, LongArrayList>> buildTermsEnum(int terms, int
 
     private ByteBuffer wrap(ByteComparable bc)
     {
-        return ByteBuffer.wrap(ByteSourceInverse.readBytes(bc.asComparableBytes(ByteComparable.Version.OSS50)));
+//        return ByteBuffer.wrap(ByteSourceInverse.readBytes(bc.asComparableBytes(ByteComparable.Version.OSS50)));
+
+        return ByteBuffer.wrap(ByteSourceInverse.readBytes(ByteSourceInverse.unescape(ByteSource.peekable(bc.asComparableBytes(ByteComparable.Version.OSS50)))));
```

Comment:
nit: clean up commented out bit

We're having to do this now because we've changed `buildStringTermsEnum()` to use `ByteComparable.of()`, which implicitly escapes? I'm trying to remember why we escape at all if we don't index multiple `ByteSource`s...

Reply:
i.e. I guess this matches what `RamBufferSegmentBuilder` does, given `asComparableBytes()` also escapes, but again, not sure if the escaping matters.

Reply:
The main reason is (as I understand it) is that the `Trie` does not know the difference between a BC terminated with a `0` (escape character) and a BC ending in a `0`.